classes/program.rb

```diff
@@ -0,0 +1,250 @@
+    case acc_menu_sel
+    # Account Menu option 1 indicates the user wants to check their balance.
+    when 1
+      puts "Your current account balance is: $#{@user.balance}"
+      puts ""
+    # Account Menu option 2 indicates the user would like to withdraw funds
+    when 2
+      # Make a withdrawal
+      withdraw_funds()
+    # Account Menu option 3 indicates the user would like to log out
+    when 3
+      @show_acc_menu = false
+      puts "Returning to the Main Menu."
+      puts ""
+    # If menu options 1-3 have failed, the world might end. Display an error
+    else
+      display_error()
+    end
+  end
+
+  # This method handles the withdrawing funds logic.
+  def withdraw_funds
+    # The response hash is used to simplify the testing on this method
+    r = {}
+    # Ask the user how much they would like to withdraw from their account
+    amount = get_input("How much would you like to withdraw?").to_i
+    # Let's make sure the account and the ATM can handle this transaction
+    user_can_withdraw = @user.can_withdraw?(amount)
+    atm_can_withdraw = @atm.can_withdraw?(amount)
+    if user_can_withdraw && atm_can_withdraw
+      # Passed that check. Now update account amounts in memory and on file...
+      @user.deduct(amount)
+      @atm.deduct(amount)
+      update_data_file(@data_file)
+      # ... and dispense the monies.
+      puts "Sure thing. Dispensing cash below..."
+      puts ""
+      puts "Your new balance is $#{@user.balance}"
+      puts ""
+    elsif amount <= 0
+      puts "You can not withdraw a negative amount."
+      puts ""
+      r[:error] = "Can not withdraw a negative amount."
+    elsif !user_can_withdraw
+      puts "You have asked to withdraw more than you have in your account"
+      puts ""
+      r[:error] = "User does not have sufficient funds"
+    elsif !atm_can_withdraw
+      puts "ATM has insufficient balance to fulfill this request."
+      puts ""
+      r[:error] = "ATM does not have sufficient funds"
+    else
+      display_error()
+      r[:error] = "Unkown error"
+    end
+    r
+  end
+
+  # This method handles parsing the users.csv file that serves as our database table.
+  def parse_data_file(file_name)
+    CSV.foreach(file_name, headers: true) do |row|
+      user = User.new
+      user.name = row["Name"]
+      user.pin = row["Pin"]
+      user.balance = row["Balance"].to_i
+      @account_holders.push(user)
+    end
+    @account_holders
+  end
+
+  # This method handles updating our user data file after a transaction to ensure the
+  # file contains the most up to date information.
+  def update_data_file(file_name)
+    header = ["Name", "Pin", "Balance"]
+    CSV.open(file_name, "wb") do |csv|
+      csv << header
+      @account_holders.each do |user|
+        data = [user.name, user.pin, user.balance]
+        csv << data
+      end
+    end
+  end
+
+  # Extract gets.chomp for easier testing purposes.
+  def chomp
+    var = gets.chomp
```

Comment:
what happens if they put in empty spaces? Perhaps a `.strip` call would be useful here